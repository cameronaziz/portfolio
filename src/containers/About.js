import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withSiteData } from 'react-static';
import Close from '../components/Close';
import Article from '../components/Article';

class About extends Component {
  render() {
    const { history, closeArticle } = this.props;
    return (
      <Article id="about" className="active" width="40rem" title="About Me">
        <span className="image main">
          <img src="/images/about.png" alt="" />
        </span>
        <p>
          My passion for open source technologies has allowed me to take
          products and applications from conception to production. I am a
          trusted communicator between technical and non-technical staff which
          allows me to architect solutions that are both rapid to deploy and
          easily scalable. I am a senior frontend engineer specializing in
          Typescript, JavaScript, React and GraphQL. I have worked on devops,
          backend and frontend teams individually and I enjoy learning new web
          technologies in my free time.
          <br /><br />
          By the way, check out my <a href="/work">awesome work</a>.
        </p>
        <Close closeArticle={closeArticle} history={history} />
      </Article>
    );
  }
}

About.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  closeArticle: PropTypes.func.isRequired,
};

export default withSiteData(About);
