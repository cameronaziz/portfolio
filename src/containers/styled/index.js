import styled from 'styled-components';
// import * as work from './work';


export const ArticleTitle = styled.h2`
  border-bottom: solid 1px #ffffff;
  width: -moz-max-content;
  width: -webkit-max-content;
  width: -ms-max-content;
  width: max-content;
  padding-bottom: 0.5rem;
  margin: 0 0 2rem 0;
`;


export default {
  ArticleTitle,
};
