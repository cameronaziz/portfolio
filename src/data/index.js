export { default as projects } from './projects';
export { default as libraries } from './libraries';