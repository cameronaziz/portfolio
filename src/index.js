import React from 'react';
import ReactDOM from 'react-dom';
import AppComponent from './App';


if (typeof document !== 'undefined') {
  const renderMethod = module.hot
    ? ReactDOM.render
    : ReactDOM.hydrate || ReactDOM.render;
  const render = Comp => {
    renderMethod(<Comp />, document.getElementById('root'));
  };
  render(AppComponent);
}

export default AppComponent;
