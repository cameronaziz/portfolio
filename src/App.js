import React, { Component } from 'react';
import { Router, Route, Switch } from 'react-static';
import { hot } from 'react-hot-loader';
import { ApolloProvider } from 'react-apollo';
import { Provider } from 'react-redux';
import Header from './components/Header';
import Footer from './components/Footer';
import client from './connectors/apollo';
import store from './connectors/redux';
import * as Containers from './containers';

import './assets/css/app.css';

export class App extends Component {
  state = { articleOpen: false }

  componentDidMount() {
    const { pathname } = window.location;
    if (pathname !== '/') {
      this.toggleArticle();
    }
  }

  toggleArticle = () => {
    const { articleOpen } = this.state;
    this.setState({
      articleOpen: !articleOpen,
    });
  }

  render() {
    const { articleOpen } = this.state;
    const mainStyle = articleOpen ? undefined : { display: 'none' };
    return (
      <ApolloProvider client={client}>
        <Provider store={store}>
          <Router>
            <div id="wrapper">
              <Header openArticle={this.toggleArticle} articleOpen={articleOpen} />
              <div id="main" style={mainStyle}>
                <Switch>
                  <Route exact path="/" />
                  <Route path="/about" render={(props) => <Containers.About {...props} closeArticle={this.toggleArticle} />} />
                  <Route path="/contact" render={(props) => <Containers.Contact {...props} closeArticle={this.toggleArticle} />} />
                  <Route path="/work" render={(props) => <Containers.Work {...props} closeArticle={this.toggleArticle} />} />
                  <Route render={(props) => <Containers.FourOhFour {...props} />} />
                </Switch>
              </div>
              <Footer />
            </div>
          </Router>
        </Provider>
      </ApolloProvider>
    );
  }
}

export default hot(module)(App);
