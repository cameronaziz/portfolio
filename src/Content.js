import React, { Component, createRef } from 'react';
import { Route, Switch, withRouter } from 'react-static';
import { Helmet } from 'react-helmet';
import Header from './components/Header';
import Footer from './components/Footer';
import * as Containers from './containers';

class Content extends Component {
  constructor(props) {
    super(props);
    if (this.props.articleOpen) {
      this.createEscapeListener();
    }
    this.container = createRef();
    this.state = { articleOpen: false, introTitle: '' };
  }

  componentDidMount() {
    const { pathname } = window.location;
    if (pathname !== '/') {
      this.toggleArticle();
    }
    this.changer = setTimeout(() => {
      const logoTitle = "Hi. I'm Cameron Aziz.";
      let randomTitle = '';
      const possible = '-+*/|aspMPM123-9AD}{[]~\\":;?/.><=+-_)(*&^%$#@!)}';
      for (let i = 0; i < logoTitle.length + 1; i += 1) {
        randomTitle = logoTitle.substr(0, i);
        for (let j = i; j < logoTitle.length; j += 1) {
          randomTitle += possible.charAt(
            Math.floor(Math.random() * possible.length),
          );
        }
        this.setRandomTitle(i, randomTitle);
        randomTitle = '';
      }
    }, 500);
  }

  componentWillUnmount = () => {
    clearTimeout(this.changer);
    clearTimeout(this.generator);
  };

  setRandomTitle = (i, introTitle) => {
    this.generator = setTimeout(() => {
      this.setState({
        introTitle,
      });
    }, i * 130);
  }

  createEscapeListener = () => {
    document.addEventListener('keydown', this.handleKeyDown);
  }

  destoryEscapeListener = () => {
    document.removeEventListener('keydown', this.handleKeyDown);
  }

  toggleArticle = (contactName) => {
    if (typeof(contactName) === 'string') {
      this.setNewTitle(contactName);
    }
    const { articleOpen } = this.state;
    this.setState({
      articleOpen: !articleOpen,
    });
    if (articleOpen) {
      this.props.history.push('/');
      this.destoryEscapeListener();
    } else {
      this.createEscapeListener();
    }
  }

  setNewTitle = (contactName) => {
    this.setState({
      introTitle: `Well hello, ${contactName}. I'll be in touch.`,
    });
  }


  handleKeyDown = (event) => {
    if (event.keyCode === 27 && this.state.articleOpen) {
      this.toggleArticle();
    }
  }

  handleOutsideClick = (event) => {
    if (event.target === this.container.current && this.state.articleOpen) {
      this.toggleArticle();
    }
  }

  render() {
    const { articleOpen, introTitle } = this.state;
    const mainStyle = articleOpen ? undefined : { display: 'none' };
    return (
      <div
        ref={this.container}
        role="button"
        tabIndex={-1}
        onClick={this.handleOutsideClick}
        onKeyDown={this.handleOutsideClick}
        id="wrapper"
      >
        <Helmet>
          <meta charSet="utf-8" />
          <title>Cameron Aziz - Javascript Engineer</title>
          <link rel="apple-touch-icon" sizes="57x57" href="/images/apple-icon-57x57.png" />
          <link rel="apple-touch-icon" sizes="60x60" href="/images/apple-icon-60x60.png" />
          <link rel="apple-touch-icon" sizes="72x72" href="/images/apple-icon-72x72.png" />
          <link rel="apple-touch-icon" sizes="76x76" href="/images/apple-icon-76x76.png" />
          <link rel="apple-touch-icon" sizes="114x114" href="/images/apple-icon-114x114.png" />
          <link rel="apple-touch-icon" sizes="120x120" href="/images/apple-icon-120x120.png" />
          <link rel="apple-touch-icon" sizes="144x144" href="/images/apple-icon-144x144.png" />
          <link rel="apple-touch-icon" sizes="152x152" href="/images/apple-icon-152x152.png" />
          <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-icon-180x180.png" />
          <link rel="icon" type="image/png" sizes="192x192"  href="/images/android-icon-192x192.png" />
          <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32.png" />
          <link rel="icon" type="image/png" sizes="96x96" href="/images/favicon-96x96.png" />
          <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16.png" />
          <link rel="manifest" href="/images/manifest.json" />
          <meta name="msapplication-TileColor" content="#ffffff" />
          <meta name="msapplication-TileImage" content="/ms-icon-144x144.png" />
        </Helmet>
        <Header introTitle={introTitle} openArticle={this.toggleArticle} articleOpen={articleOpen} />
        <div id="main" style={mainStyle}>
          <Switch>
            <Route exact path="/" />
            <Route path="/about" render={(props) => <Containers.About {...props} closeArticle={this.toggleArticle} />} />
            <Route path="/contact" render={(props) => <Containers.Contact {...props} closeArticle={this.toggleArticle} />} />
            <Route path="/work" render={(props) => <Containers.Work {...props} closeArticle={this.toggleArticle} />} />
            <Route render={(props) => <Containers.FourOhFour {...props} />} />
          </Switch>
        </div>
        <Footer />
      </div>
    );
  }
}

export default withRouter(Content);
