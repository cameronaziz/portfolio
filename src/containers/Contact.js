import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withSiteData } from 'react-static';
import Close from '../components/Close';
import Article from '../components/Article';
import { ArticleTitle } from './styled';

class Contact extends Component {
  render() {
    const { history, closeArticle } = this.props;
    return (
      <Article id="contact" className="active">
        <ArticleTitle>Contact</ArticleTitle>
        <form method="post" action="#">
          <div className="fields">
            <div className="field half">
              <label htmlFor="name">Name</label>
              <input type="text" name="name" id="name" />
            </div>
            <div className="field half">
              <label htmlFor="email">Email</label>
              <input type="text" name="email" id="email" />
            </div>
            <div className="field">
              <label htmlFor="message">Message</label>
              <textarea name="message" id="message" rows="4" />
            </div>
          </div>
          <ul className="actions">
            <li><input type="submit" value="Send Message" className="primary" /></li>
            <li><input type="reset" value="Reset" /></li>
            <li>
              <ul className="icons">
                <li><a href="#" className="icon fa-linkedin"><span className="label">Facebook</span></a></li>
                <li><a href="#" className="icon fa-github"><span className="label">GitHub</span></a></li>
              </ul>
            </li>
          </ul>
        </form>

        <Close closeArticle={closeArticle} history={history} />
      </Article>
    );
  }
}

Contact.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  closeArticle: PropTypes.func.isRequired,
};

export default withSiteData(Contact);
