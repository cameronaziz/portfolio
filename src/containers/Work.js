import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withSiteData } from 'react-static';
import Close from '../components/Close';
import Card from '../components/Card';
import { projects } from '../data';
import Article from '../components/Article';
import { ArticleTitle } from './styled';

class Work extends Component {
  render() {
    console.log(this.props);
    const { history, closeArticle } = this.props;
    return (
      <Article id="work" className="active">
        <ArticleTitle>Work</ArticleTitle>
        {/* <span className="image main">
          <img src="/images/img_avatar.png" alt="" />
        </span> */}
        {projects.map(project => <Card card={project} />)}
        <Close closeArticle={closeArticle} history={history} />
      </Article>
    );
  }
}

Work.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  closeArticle: PropTypes.func.isRequired,
};

export default withSiteData(Work);
