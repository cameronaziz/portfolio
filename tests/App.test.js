import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-static';
import './testsConfig';

import * as App from '../src/App';
import * as Containers from '../src/containers';

describe('App Component', () => {
  it('will match snapshot', () => {
    const component = renderer.create(<App.App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('will provide 404 page for bad path', () => {
    console.log(App);
    const wrapper = mount(
      <MemoryRouter initialEntries={['/random']}>
        <App.App />
      </MemoryRouter>,
    );
    expect(wrapper.find(Containers.FourOhFour)).toHaveLength(1);
  });
});
