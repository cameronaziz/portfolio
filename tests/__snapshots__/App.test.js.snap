// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component will match snapshot 1`] = `
<div
  id="wrapper"
>
  <header
    id="header"
  >
    <div
      className="logo"
    >
      <span
        className="icon fa-file-code-o"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="inner"
      >
        <h1>
          Hi. I'm Cameron Aziz.
        </h1>
        <p>
          I'm a web engineer specializing in full stack Javascript applications. Like this one.
        </p>
      </div>
    </div>
    <nav>
      <ul>
        <li>
          <a
            aria-current={null}
            href="/work"
            onClick={[Function]}
          >
            Work
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/contact"
            onClick={[Function]}
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    id="main"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <footer
    id="footer"
  >
    <p
      className="copyright"
    >
      Â© 
      2018
       Cameron Aziz
    </p>
  </footer>
</div>
`;
